runtime/src/main/java/org/corfudb/runtime/object/VersionLockedObject.java

```diff
@@ -672,25 +674,32 @@ protected void syncStreamUnsafe(ISMRStream stream, long timestamp) {
         log.trace("Sync[{}] {}", this, (timestamp == Address.OPTIMISTIC)
                 ? "Optimistic" : "to " + timestamp);
         long syncTo = (timestamp == Address.OPTIMISTIC) ? Address.MAX : timestamp;
-        stream.streamUpTo(syncTo)
-                .forEachOrdered(entry -> {
-                    try {
-                        Object res = applyUpdateUnsafe(entry, timestamp);
-                        if (timestamp == Address.OPTIMISTIC) {
+        Timer timer = VloMetricsHelper.metrics.timer(VloMetricsHelper.VLO_SYNC_STREAM + stream.getID());
+        Histogram histogram = VloMetricsHelper.metrics.histogram(VloMetricsHelper.VLO_SYNC_COUNT + stream.getID());
+        AtomicLong counter = new AtomicLong();
```

Comment:
If you move this string concatenation outside, you don't have to do it on each access.

Reply:
VLO is protected under a lock, so you don't need to use an atomic counter.

Reply:
Moved string concatenation to the class constructor. Replaced atomic counter by a size 1 array.

Reply:
Oh, I see the motivation. I believe using an array worse idea that using Atomic. 
If you don't like to use atomics for this, just create a class
```
static class Counter {
    int counter;
}
```
For the sake of clear code should work well for you.
Or you can add that to a common module or util package, up to you